include/envoy/common/platform.h

```diff
@@ -65,6 +65,7 @@ typedef uint32_t mode_t;
 
 typedef SOCKET os_fd_t;
 typedef HANDLE filesystem_os_id_t; // NOLINT(modernize-use-using)
+typedef DWORD signal_t;            // NOLINT(modernize-use-using)
```

Comment:
Could you update the issue title to something like:

win32 signal: add term and ctrl-c signal handlers

Reply:
You mean the PR title right? The corresponding issue is a bit more broad because we also want to enable SCM (service control manager) which is the second way to manage a process in Windows (by running it as a service)

Reply:
Yeah, sorry.  I meant PR title.